File: 389-find-the-difference/find-the-difference.cpp
#include <string>

class Solution {
public:
    char findTheDifference(std::string s, std::string t) {
        char result = 0;
        
        // XOR all characters in both s and t
        for (char ch : s + t) {
            result ^= ch;
        }
        
        return result;
    }
};

File: 389-find-the-difference/find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:

        if s == "":
            return t

        ans = []
        for i in t:
            ans.append(i)
        
        for i in s:
            if i in ans:
                ans.remove(i)
        return "".join(ans)
        